---
title: "08_Motif_Analysis"
output: html_notebook
---

```{r}
library(JASPAR2024)
library(TFBSTools)
library(Biostrings)
```

Get (Arabidopsis) LFY binding site. Fern is similar, although it might be worth creating my own Fern pfm at some point.
```{r}
dbfile <- JASPAR2024()@db

pfm.lfy <- getMatrixByID(dbfile, ID = "MA0590.1")
pwm.lfy <- toPWM(pfm.lfy)
icm.lfy <- toICM(pfm.lfy)
```

load genome
```{r}
genome <- readDNAStringSet("~/Sequences/ref_genomes/C_richardii_v2.1/assembly/Crichardii_676_v2.0.fa.gz")
genome <- genome[nchar(genome) >= 100000] # only keep reasonably size scaffolds
```

load gff

```{r}
gff <- rtracklayer::import.gff("~/Sequences/ref_genomes/C_richardii_v2.1/annotation/Crichardii_676_v2.1.gene.gff3.gz") 
gff <- gff[gff$type=="mRNA" & gff$longest==1]
```

search for occurrences (search entire genome...this makes it easier to try different promoter lenghts later)

```{r, warning=FALSE}
system.time(
hits <- searchSeq(pwm.lfy, genome, strand = "+") # because the site is ~ palindromic we don't need to search both strands.
)
```

```{r}
hits
```

```{r}
hits.gr <- as(hits, "GRanges")
hits.gr
```

```{r}
promoters <- flank(gff, 5000)

overlaps <- tibble(geneID = promoters$Name,
                   count = countOverlaps(promoters, hits.gr, type = "any", ignore.strand=TRUE)
                   )
overlaps
```

## WGCNA

### gametophyte samples
```{r}
load("../output/WGCNA_gametophyte_all.Rdata")
gmodule.assignment <- module.assignment
```

### LFY1
```{r}
lfy1.gmodule <- gmodule.assignment %>% filter(str_detect(geneID, CrLFY1)) %>% pull(module)

gmodule.assignment %>%
  mutate(geneID = str_remove(geneID, "\\.v2")) %>% 
  right_join(overlaps) %>%
  mutate(module=ifelse(is.na(module), "none", module),
         in.network = module==lfy1.gmodule) %>%
  t.test(count ~ in.network, data = .)
```
```{r}
lfy2.gmodule <- gmodule.assignment %>% filter(str_detect(geneID, CrLFY2)) %>% pull(module)

gmodule.assignment %>%
  mutate(geneID = str_remove(geneID, "\\.v2")) %>% 
  right_join(overlaps) %>%
  mutate(module=ifelse(is.na(module), "none", module),
         in.network = module==lfy2.gmodule) %>%
  t.test(count ~ in.network, data = .)
```

### sporophyte samples
```{r}
load("../output/WGCNA_sporophyte_all.Rdata")
smodule.assignment <- module.assignment
```

### LFY1
```{r}
lfy1.smodule <- smodule.assignment %>% filter(str_detect(geneID, CrLFY1)) %>% pull(module)

smodule.assignment %>%
  mutate(geneID = str_remove(geneID, "\\.v2")) %>% 
  right_join(overlaps) %>%
  mutate(module=ifelse(is.na(module), "none", module),
         in.network = module==lfy1.smodule) %>%
  t.test(count ~ in.network, data = .)
```

```{r}
lfy2.smodule <- smodule.assignment %>% filter(str_detect(geneID, CrLFY2)) %>% pull(module)

smodule.assignment %>%
  mutate(geneID = str_remove(geneID, "\\.v2")) %>% 
  right_join(overlaps) %>%
  mutate(module=ifelse(is.na(module), "none", module),
         in.network = module==lfy2.smodule) %>%
  t.test(count ~ in.network, data = .)
```

## load TGCN networks

## TGCN Gametophyte

### LFY1 Cutoff 6

All genes in submodules
```{r}
read_csv("../output/TGCN_LFY1_gam/results/LFY1_gam_c6_TGCN.csv") %>% 
  rename(geneID = genes) %>% 
  right_join(overlaps) %>%
  mutate(in.network=!is.na(hubGene))  %>%
  t.test(count ~ in.network, data = .)
```
hub genes only
```{r}
read_csv("../output/TGCN_LFY1_gam/results/LFY1_gam_c6_TGCN.csv") %>% 
  select(geneID=hubGene) %>%
  distinct() %>% 
  mutate(in.network=TRUE) %>%
  right_join(overlaps) %>%
  mutate(in.network=!is.na(in.network))  %>%
  t.test(count ~ in.network, data = .)
```

### LFY2 Cutoff 5

Note: this gets more significant as I raise the cutoff (e.g. to 7)

All genes in submodules
```{r}
read_csv("../output/TGCN_LFY2_gam/results/LFY2_gam_c6_TGCN.csv") %>% 
  rename(geneID = genes) %>% 
  right_join(overlaps) %>%
  mutate(in.network=!is.na(hubGene)) %>%
  t.test(count ~ in.network, data = .)

```
hub genes only
```{r}
read_csv("../output/TGCN_LFY2_gam/results/LFY2_gam_c5_TGCN.csv") %>% 
  select(geneID=hubGene) %>%
  distinct() %>% 
  mutate(in.network=TRUE) %>%
  right_join(overlaps) %>%
  mutate(in.network=!is.na(in.network)) %>%
  t.test(count ~ in.network, data = .)
```

## TGCN Sporophyte

### LFY1 Cutoff 5

All genes in submodules
```{r}
read_csv("../output/TGCN_LFY1_sp/results/LFY1_sp_c5_TGCN.csv", show_col_types = FALSE) %>% 
  rename(geneID = genes) %>% 
  right_join(overlaps) %>%
  mutate(in.network=!is.na(hubGene))  %>%
  t.test(count ~ in.network, data = .)
```
hub genes only
```{r}
read_csv("../output/TGCN_LFY1_gam/results/LFY1_gam_c5_TGCN.csv", show_col_types = FALSE) %>% 
  select(geneID=hubGene) %>%
  distinct() %>% 
  mutate(in.network=TRUE) %>%
  right_join(overlaps) %>%
  mutate(in.network=!is.na(in.network))  %>%
  t.test(count ~ in.network, data = .)
```

### LFY2 Cutoff 4

Note: cutoff of 5 or higher is n.s.

All genes in submodules
```{r}
read_csv("../output/TGCN_LFY2_sp/results/LFY2_sp_c4_TGCN.csv", show_col_types = FALSE) %>% 
  rename(geneID = genes) %>% 
  right_join(overlaps) %>%
  mutate(in.network=!is.na(hubGene)) %>%
  t.test(count ~ in.network, data = .)

```
hub genes only
```{r}
read_csv("../output/TGCN_LFY2_sp/results/LFY2_sp_c4_TGCN.csv", show_col_types = FALSE) %>% 
  select(geneID=hubGene) %>%
  distinct() %>% 
  mutate(in.network=TRUE) %>%
  right_join(overlaps) %>%
  mutate(in.network=!is.na(in.network)) %>%
  t.test(count ~ in.network, data = .)
```


Next steps...


Also think about looking at relationship between # of LFY binding sites and correlation


