---
title: "Download and QC"
author: "Julin Maloof"
date: "2024-12-04"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
```

Get the files, generate MD5
```{bash, eval=FALSE}
# On Jetstream
cd /raw_seq_data/
sudo -s
mkdir Julin-FERN
cd Julin-FERN/
sftp  hmccon_gmail@sftp.genewiz.com
cd 30-1052403065/
md5sum 00_fastq/*gz > md5s.download.txt
cat 00_fastq/*md5 > md5s.original.txt
cd ../
chmod o+x 30-1052403065/
cd 30-1052403065/
chmod o+x 00_fastq/
```

Downloaded the V2.1 reference genome from phytozome to `/analyzed_data/ref_genomes/C_richardii_v2.1

Check MD5
```{r}
md5.download <- read_delim("../input/md5s.download.txt", 
                           col_names = c("md5.download", "X", "file"), 
                           col_select = c(md5.download, file)) %>%
    mutate(file = basename(file))



md5.original <- read_delim("../input/md5s.original.txt",
                           col_names = c("md5.original", "X", "file"),
                           col_select = c(md5.original, file)) %>%
  mutate(file = basename(file))
```
```{r}
full_join(md5.download, md5.original) %>%
  summarize(all(md5.download==md5.original))
```




Run fastqc

```{bash}
cd ../output
mkdir fastqc
fastqc -o fastqc --noextract --threads 4 /raw_seq_data/Julin-FERN/30-1052403065/00_fastq/*gz
```

Run multiqc

```{bash}
conda activate sequencing
multiqc -o multiqc fastqc/
```

